### ISSUE TITLE
`ModuleMap`'s handling of devices

### ISSUE THREAD

**Issue by frostedoyster** [lab-cosmo/metatensor#554]
At the moment, `ModuleMap` takes a device at initialization and there is no way to change it later. However, this does not correspond to idiomatic torch, where modules can be moved from and to devices. Since `Labels` cannot be registered as part of a `torch.nn.Module`, I think the only possibility for now is to move the `Labels` to the correct device during the forward function. Another option would be to override the `to` function of `ModuleMap` so the `Labels` are moved as well, but I do not know how difficult this would be.
@agoscinski @jwa7 @abmazitov @Luthaf

**Comment by Luthaf** [comment]
I think this is what https://github.com/lab-cosmo/metatensor/pull/496 was trying to fix?
